File: fibo.c
#include <stdio.h>
#include <stdlib.h>

unsigned long fibo(unsigned long n) {
	if (n < 2) {
		return n;
	} else {
		return fibo(n - 1) + fibo(n - 2);
	}
}

int main(int argc, char **argv) {
	printf("%ld\n", fibo(atol(argv[1])));
	return 0;
}

File: fibo.ml
let rec fibo n =
  if n < 2 then n
  else fibo (n - 2) + fibo (n - 1)

let _ =
  let n =
    try int_of_string Sys.argv.(1)
    with Invalid_argument _ -> 1 in
  Printf.printf "%d\n" (fibo n)
